src/mixins/auth-api-mixin.ts

```diff
@@ -16,23 +15,95 @@ export default class AuthApiMixin extends Vue {
+
+    return axios.get(url, config)
+  }
+
+  /**
+   * Fetches the current user data.
+   * @returns a promise to return the data
+   */
+  async fetchCurrentUser (): Promise<any> {
+    const authUrl = sessionStorage.getItem('AUTH_API_URL')
+    const config = { baseURL: authUrl }
+
+    return axios.get('users/@me', config)
+  }
+
+  /**
+   * Fetches the auth info of the current business.
+   * @returns a promise to return the data
+   */
+  async fetchAuthInfo (): Promise<AuthInformationIF> {
+    if (!this.getBusinessId) throw new Error('Invalid business id')
+
+    const url = `entities/${this.getBusinessId}`
+    const authUrl = sessionStorage.getItem('AUTH_API_URL')
+    const config = { baseURL: authUrl }
+
+    return axios.get(url, config)
+      .then(response => {
+        if (response?.data) {
+          return {
+            contacts: response.data.contacts,
+            folioNumber: response.data.folioNumber
+          }
+        } else {
+          // eslint-disable-next-line no-console
+          console.log('fetchAuthInfo() error - invalid response =', response)
+          throw new Error('Invalid API response')
+        }
+      })
+  }
+
+  /**
+   * Updates the businesses contact information.
+   * @param contactInfo the contact information object
    */
   updateContactInfo (contactInfo: ContactPointIF): Promise<any> {
-    try {
-      const contactPayload = {
-        email: contactInfo.email,
-        phone: contactInfo.phone,
-        phoneExtension: contactInfo.extension
-      }
-      const authUrl = sessionStorage.getItem('AUTH_API_URL')
-      let url = `${authUrl}entities/${this.getBusinessId}/contacts`
-
-      return axios.put(url, contactPayload)
-    } catch (error) {
-      // TODO: Update error handling modal after design review
-      this.$root.$emit('save-error-event', error)
+    if (!this.getBusinessId) throw new Error('Invalid business id')
+
+    const data = {
+      email: contactInfo.email,
+      phone: contactInfo.phone,
+      phoneExtension: contactInfo.extension
     }
+    const authUrl = sessionStorage.getItem('AUTH_API_URL')
+    let url = `${authUrl}entities2/${this.getBusinessId}/contacts`
+
+    return axios.put(url, data)
+      .catch(error => {
+        this.$root.$emit('update-error-event', 'Failed to update Contact Info')
+        throw error
+      })
+  }
+
+  /**
+   * Updates the folio number.
+   * @param folioNumber the folio number
+   */
+  updateFolioNumber (folioNumber: string): Promise<any> {
+    const businessIdentifier = this.getBusinessId
+    if (!businessIdentifier) throw new Error('Invalid business id')
+
+    const data = { businessIdentifier, folioNumber }
+    const authUrl = sessionStorage.getItem('AUTH_API_URL')
+    let url = `${authUrl}entities/${businessIdentifier}`
+
+    return axios.patch(url, data)
+      .catch(error => {
+        this.$root.$emit('update-error-event', 'Failed to update Folio Number')
+        throw error
```

Comment:
This could be done other ways, but this works.
1. this method emits an event to display the error dialog
2. the calling method catches the error and reset its data